jquery.flot.js

```diff
@@ -1670,40 +1693,94 @@ Licensed under the MIT license.
             x = edges.x;
             y = edges.y;
 
-            // draw major tick marks
+			// draw major tick marks
             ctx.strokeStyle = axis.options.color;
             ctx.beginPath();
+
             for (i = 0; i < axis.ticks.length; ++i) {
                 var v = axis.ticks[i].v,
                     xoff = 0,
-                    yoff = 0;
+                    yoff = 0;           
 
-                if (isNaN(v) || v < axis.min || v > axis.max)
-                    continue;
+                if (!isNaN(v) && v >= axis.min && v <= axis.max) {       
+                    if (axis.direction == "x") {
+                        x = axis.p2c(v);
+                        yoff = t;
+                        
+                        if (axis.position == "top") {
+                            yoff = -yoff;
+                        }
+                    } else {
+                        y = axis.p2c(v);
+                        xoff = t;
 
-                if (axis.direction == "x") {
-                    x = axis.p2c(v);
-                    yoff = t;
+                        if (axis.position == "left") {
+                            xoff = -xoff;
+                        }
+                    }
 
-                    if (axis.position == "top")
-                        yoff = -yoff;
-                } else {
-                    y = axis.p2c(v);
-                    xoff = t;
+                    if (ctx.lineWidth == 1) {
+                        if (axis.direction == "x")
+                            x = Math.floor(x) + 0.5;
+                        else
+                            y = Math.floor(y) + 0.5;
+                    }
 
-                    if (axis.position == "left")
-                        xoff = -xoff;
-                }
+                    ctx.moveTo(x, y);
+                    ctx.lineTo(x + xoff, y + yoff);
+                }
+
+                //draw minor tick marks
+                if(minorTicks == true && i < axis.ticks.length - 1)
+                {
+                    var minorTicksNr = 4; // default number of minor ticks
```

Comment:
Move this constant and the default tick length to jquery.flot.uiConstants.js file